src/frontend/components/UI/LogFileUploadDialog/index.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { CircularProgress } from '@mui/material'
+
+import { Dialog, DialogContent, DialogFooter } from '../Dialog'
+import { useShallowGlobalState } from 'frontend/state/GlobalStateV2'
+
+export default function LogUploadDialog() {
+  const { t } = useTranslation()
+  const { uploadLogFileProps, setUploadLogFileProps } = useShallowGlobalState(
+    'uploadLogFileProps',
+    'setUploadLogFileProps'
+  )
+
+  const [confirmed, setConfirmed] = useState(false)
+  const [uploading, setUploading] = useState(false)
+  const [error, setError] = useState(false)
+  const [uploadUrl, setUploadUrl] = useState<string | null>(null)
+
+  const onClose = useCallback(() => {
+    setConfirmed(false)
+    setUploading(false)
+    setUploadLogFileProps(false)
+  }, [setConfirmed, setUploading, setUploadLogFileProps])
+
+  const doUpload = useCallback(async () => {
+    if (!uploadLogFileProps) return
+
+    setConfirmed(true)
+    setUploading(true)
+
+    const result = await window.api.uploadLogFile(
+      uploadLogFileProps.name,
+      uploadLogFileProps.appNameOrRunner
+    )
+
+    setUploading(false)
+    if (!result) {
+      setError(true)
+      return
+    }
+    const [url] = result
+    await navigator.clipboard.writeText(url)
+    setUploadUrl(url)
+  }, [uploadLogFileProps])
+
+  const [dialogContent, dialogFooter] = useMemo(() => {
+    if (!uploadLogFileProps) return ['', '']
+    if (!confirmed)
+      return [
+        t(
+          'setting.log.upload.confirm',
+          'Do you really want to upload "{{name}}"?',
+          { name: uploadLogFileProps.name }
+        ),
+        <>
+          <button onClick={doUpload} className={'button is-primary'}>
+            {t('box.yes')}
+          </button>
+          <button onClick={onClose} className={'button is-danger'}>
+            {t('box.no')}
+          </button>
+        </>
+      ]
+    if (uploading)
+      return [
+        <>
+          <CircularProgress />
```

Comment:
```suggestion
          <CircularProgress />
          <br />
```

the loader looks weird in the same line as the text